packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -677,9 +677,13 @@ function updateOffscreenComponent(
   const prevState: OffscreenState | null =
     current !== null ? current.memoizedState : null;
 
+  markRef(current, workInProgress);
+
   if (
     nextProps.mode === 'hidden' ||
-    (enableLegacyHidden && nextProps.mode === 'unstable-defer-without-hiding')
+    (enableLegacyHidden &&
+      nextProps.mode === 'unstable-defer-without-hiding') ||
+    workInProgress.stateNode._isDetached
```

Comment:
This might be a race condition. StateNode is shared. It is a shared mutable state.

Reply:
Add a comment that this needs some more thinking.